### PR TITLE
Bootstrap using a local directory

### PR DESCRIPTION
**Pull request by lrytz**
I'm testing this change in the 2.13.x-new-collections branch.

Added the WIP label as maybe there will be more changes, and I want to add a small script for local bootstrapping.

### CODE REVIEW

**Comment by lrytz** [comment]
@retronym since you voiced interest in a local bootstrap script: what did you have in mind?
  - follow the same procedure a bootstrap on travis, including modules, but `publishLocal` in the quick stage (instead of publishing to scala-integration)?
  - or a script for `sbt publishLocal; sbt -Dstarr.version=... dist/mkQuick`?

what use cases do you have in mind?

**Comment by lrytz** [comment]
Pushed a first version of the local bootstrap script that follows what we do on CI.

File: scripts/bootstrap_local [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+source scripts/common
+source scripts/bootstrap_fun
+
+travis_fold_start() { :; }
+travis_fold_end() { :; }
+
+IVY2_DIR="$HOME/.ivy2"
+# The quick stage `publish`es to `integrationRepoUrl`. This makes it into a `publishLocal`.
+integrationRepoUrl="file://$IVY2_DIR/local"
```
Comment (lrytz) [inline scripts/bootstrap_local]: this doesn't work as intended, as it publishes there in maven format, but should be ivy format. need do actually use `publishLocal`.

**Comment by retronym** [comment]
A convenience wrapper around `sbt publishLocal; sbt -Dstarr.version=... dist/mkQuick` is what I had in mind. Maybe it should even just be `sbt setupPublishCore publishLocal updateVersionsProperties`, after which we can just run `sbt whatever` and be in bootstrapped mode.

It occurs to me after #6479 that we should really be using bootstrapped mode for local development all the time to dogfood recent changes to the compiler. We'll get a better idea of bugs that we have poor testing for (e.g. interaction with SBT)

**Comment by lrytz** [comment]
Merging the changes without the local script for now. https://github.com/scala/scala/commits/2.13.x-new-collections contains the equivalent changes for 2.13.x.

**Pull request merged**